src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -1315,15 +1312,6 @@ export class SavedObjectsRepository {
         _source_includes: ['namespace', 'namespaces', 'originId'],
         require_alias: true,
       })
-      .then((res) => {
```

Comment:
We're not ignoring 404's and they are [caught](https://github.com/elastic/kibana/blob/master/src/core/server/saved_objects/service/lib/repository_es_client.ts#L41) from `retryCallCluster` calls.

`decorateEsError` doesn't pass the response headers to the decorated error, so the product header check needs to be done in https://github.com/elastic/kibana/blob/master/src/core/server/saved_objects/service/lib/repository_es_client.ts#L41